public/video-ui/src/components/VideoEdit/VideoEdit.js

```diff
@@ -1,87 +1,116 @@
 import React from 'react';
-import VideoTitleEdit from './formComponents/VideoTitle';
-import VideoDescriptionEdit from './formComponents/VideoDescription';
-import VideoCategorySelect from './formComponents/VideoCategory';
-import VideoPosterEdit from './formComponents/VideoPoster';
-import YoutubeCategorySelect from './formComponents/YoutubeCategory';
-import VideoExpiryEdit from './formComponents/VideoExpiry';
-import YoutubeChannelSelect from './formComponents/YoutubeChannel';
-import PrivacyStatusSelect from './formComponents/PrivacyStatus';
-import {validate} from '../../constants/videoEditValidation';
-import { Field, reduxForm } from 'redux-form';
+import {ManagedForm, ManagedField} from '../ManagedForm';
+import TextInput from '../FormFields/TextInput';
+import TextArea from '../FormFields/TextArea';
+import SelectBox from '../FormFields/SelectBox';
+import DatePicker from '../FormFields/DatePicker';
+import {fieldLengths} from '../../constants/videoEditValidation';
+import {videoCategories} from '../../constants/videoCategories';
+import { privacyStates } from '../../constants/privacyStates';
+import ImageSelector from '../FormFields/ImageSelector';
 
-const VideoEdit = (props) => {
+class VideoEdit extends React.Component {
 
-      return (
-        <div>
-          <Field
-            name="title"
-            type="text"
-            component={VideoTitleEdit}
-            video={props.video}
-            updateVideo={props.updateVideo}
-            editable={props.editable} />
+  hasCategories = () => this.props.youtube.categories.length !== 0;
+  hasChannels = () => this.props.youtube.channels.length !== 0;
 
-          <Field
-            name="description"
-            type="text"
-            component={VideoDescriptionEdit}
-            video={props.video}
-            updateVideo={props.updateVideo}
-            editable={props.editable} />
+  componentWillMount() {
+    if (! this.hasCategories()) {
+      this.props.youtubeActions.getCategories();
+    }
+    if (! this.hasChannels()) {
+      this.props.youtubeActions.getChannels();
+    }
+  }
 
-          <Field
-            name="category"
-            type="select"
-            component={VideoCategorySelect}
-            video={props.video}
-            updateVideo={props.updateVideo}
-            editable={props.editable} />
+  render() {
 
-          <Field
-            name="posterImage"
-            component={VideoPosterEdit}
-            video={props.video}
-            editable={props.editable}
-            updateVideo={props.updateVideo} />
+    return (
+      <div className="form__group">
+        <ManagedForm
+          data={this.props.video}
+          updateData={this.props.updateVideo}
+          editable={this.props.editable}
+          updateFormErrors={this.props.updateFormErrors}
+        >
+          <ManagedField
```

Comment:
For me it would be cleaner to have single elements for each field type rather than nesting them, since they're all managed. So you could say something like:

```html
<TextInput field="title" name="Title" maxLength={fieldLengths.title} isRequired />
```

but not a deal breaker